src/univariate/continuous/cosine.jl

```diff
@@ -33,32 +45,34 @@ mode(d::Cosine) = d.μ
 
 var(d::Cosine) = d.σ^2 * 0.13069096604865779  # 0.130... = 1/3 - 2 / π^2
 
-skewness(d::Cosine) = 0.0
+skewness{T<:Real}(d::Cosine{T}) = zero(T)
 
-kurtosis(d::Cosine) = -0.59376287559828102362
+kurtosis{T<:Real}(d::Cosine{T}) = -0.59376287559828102362*one(T)
 
 
 #### Evaluation
 
-function pdf(d::Cosine, x::Float64)
+function pdf{T<:Real}(d::Cosine{T}, x::Real)
     if insupport(d, x)
         z = (x - d.μ) / d.σ
-        return (1.0 + cospi(z)) / (2 * d.σ)
+        return (1 + cospi(z)) / (2d.σ)
     else
-        return 0.0
+        return zero(T)
```

Comment:
I realize that some of these will be type unstable for integers and some will fail when a conversion to integer is not possible. I'm not sure how important these cases will be.